### ISSUE TITLE
XML comments

### ISSUE THREAD

**Issue by mikebeaton** [Giannoudis/TimePeriodLibrary#2]
Hi,

This library could really do with XML comments added, and then baked into the NuGet library to give full IntelliSense (e.g. when I load up your NuGet library and then try to instantiate my first `DateDiff` object, there is no IntelliSense to tell me whether the first or second `DateTime` object is the start date).

I just had to add a load of XML comments to a NuGet library I am maintaining, so I appreciate that it's not the most fun job - but I do agree (with the person who raised the issue in my library!) that it's definitely worth doing for a project which you might hope that lot of other people will use.

Thanks!

**Comment by Nirntendo** [comment]
Yes, this library really needs XML comments. It's a needle in a haystack if you're confronted with it for the first time.

**Comment by mikebeaton** [comment]
Just in case it helps, on my own project I had IntelliSense XML comments in the code but I couldn't figure out why people were still having problems with no IntelliSense on the NuGet package - then I came across this which explained the problem, and how to fix it: [stackoverflow.com/a/57731750/795690](https://stackoverflow.com/a/57731750/795690)

I have not had time to double-check as to whether this could actually be the issue for your project before posting, I hope that's okay, but just in case this is relevant...

Once again thanks anyway for what's otherwise obviously a genuinely useful library!
